chia/wallet/wallet_state_manager.py

```diff
@@ -921,7 +921,7 @@ async def handle_nft(
                         self.log.info(f"No NFT, deleting wallet {nft_wallet.did_id.hex()} ...")
                         await self.user_store.delete_wallet(nft_wallet.wallet_info.id)
                         self.wallets.pop(nft_wallet.wallet_info.id)
-            if nft_wallet_info.did_id == new_did_id:
+            if nft_wallet_info.did_id == new_did_id and new_derivation_record is not None:
```

Comment:
what's the reasoning behind all these derivation record checks?

Reply:
This is for handling the unexpected wallet creation/deletion issue.